src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -305,32 +305,30 @@ private static ulong ConvertBigIntegerToFloatingPointBits(ref BigInteger value,
         }
 
         // get 32-bit integer from at most 9 digits
-        private static uint DigitsToUInt32(byte* p, int count)
+        private static uint DigitsToUInt32(Span<byte> p, int count)
```

Comment:
This should be changed to:
```csharp
    private static uint DigitsToUInt32(Span<byte> p)
    {
        Debug.Assert((1 <= p.Length) && (p.Length <= 9));

        uint res = 0;

        for (int i = 0; i < p.Length; i++)
        {
            res = (10 * res) + (uint)(p[i] - '0');
        }

        return res;
    }
```

The following shows the codegen difference with bounds checks being elided when the loop upper limit is the span length:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBEBLbAcwDsJcGDmFwBuGjWIBmJqQYBhBgG8aDNQwAOUDgDdsGGE3JIGAVw68MDTtw4ZcAFQgBVAJKWppABQBlDdl4AHmAATwMAPk00BgsrSFNLAEpVdRVqdQzrGGBTbhYAQVxcGCgMLy9yBkCAXgZ4pIYAMkaGL3qrGoYATkTE8WoUzLNYhlhcBlqvcySvDQBtOgBdBjgGAHI6Nb6JdKGGADNoVpGOCYZyURiquogEjEuOAGpH5N2htL3MsbOKxgAqUYwXCJBiPTRzDjLVYbNb9T4AXx2n2IAHZAWJBgxEW8GJitLp9IZmCZplYbHZHC53BhPL5/EFQhFNK8Mh89mxsrkCkUSmUKlVahoWAAZGC8bgYAAWIOarSFovFUoF3V6/UxGVJ6LOdDVOIyhygx0sV1qOqugU0IrFEslD2eLL2bM+am+k3I/3RILBU1iiVmEKh6022z16mx6vUqPRcLDNHhQA==

Reply:
Same goes for `DigitsToUInt64`. These are two of the hottest loops and avoiding the bounds checks is fairly critical.
The additional slice to get `p` to the right length (`input.Slice(0, count)`) and the handling of `0` in the loop causing a "0 * 10" is much less than the bounds checks and additional jump every iteration.

Reply:
Done